R/run-nmtran.R

```diff
@@ -0,0 +1,390 @@
+      nm_vers <- paste(names(default_nm), collapse = ", ")
+      cli::cli_abort(
+        c(
+          "Found multiple default NONMEM versions ({nm_vers}) at {.path {bbi_yaml_path}}",
+          "i" = "Please ensure only one version is set to the default"
+        )
+      )
+    }else if(length(default_nm) == 1){
+      default_nm <- default_nm[[1]]
+    }else{
+      # If no default, use the last one (likely higher version)
+      default_nm <- nm_config[[length(nm_config)]]
+    }
+  }
+
+  # Set NM-TRAN executable path
+  nm_path <- default_nm$home
+  nmtran_exe <- file.path(nm_path, "tr", "NMTRAN.exe")
+
+  # Check if nmtran_presort exists
+  nmtran_presort_exe <- file.path(nm_path, "util", "nmtran_presort")
+  if(ON_WINDOWS) nmtran_presort_exe <- paste0(nmtran_presort_exe, ".exe")
+  run_presort <- unname(fs::file_exists(nmtran_presort_exe))
+
+  return(
+    list(
+      nmtran_exe = nmtran_exe,
+      nonmem_version = basename(nm_path),
+      cmd_args = cmd_args,
+      nmtran_presort_exe = if(run_presort) nmtran_presort_exe else NULL
+    )
+  )
+}
+
+#' Execute `NM-TRAN` in a given directory
+#'
+#' Execute `NM-TRAN` in a given directory. Also runs `nmtran_presort` if an
+#' executable is found. This assumes the model at `mod_path` has already been
+#' copied to the top-level of `dir`.
+#'
+#' @param nmtran_exe Path to `NM-TRAN` executable.
+#' @param mod_path Path of a model to evaluate. Should be relative to `dir`.
+#' @param cmd_args A character vector of command line arguments for the
+#'   `NM-TRAN` execution call
+#' @param nmtran_presort_exe Path to `nmtran_presort` executable. If provided,
+#'   will run `nmtran_presort` ahead of `NM-TRAN` for `NONMEM 7.4` and later.
+#'   Set to `NULL` to skip this step.
+#' @param dir Directory in which to execute the command.
+#'
+#' @keywords internal
+execute_nmtran <- function(
+    nmtran_exe,
+    mod_path,
+    cmd_args = c("0", "0", "2"),
+    nmtran_presort_exe = NULL,
+    dir = "."
+){
+  checkmate::assert_directory_exists(dir)
+  checkmate::assert_character(cmd_args, len = 3)
+
+  run_dir <- as.character(fs::path_real(dir))
+  nmtran_input <- mod_path_new <- file.path(run_dir, basename(mod_path))
+
+  # This should only be possible via manual intervention
+  if(!fs::file_exists(mod_path_new)){
+    cli::cli_abort("Could not find model at {.path {mod_path_new}}")
+  }
+
+  # Preprocess with nmtran_presort
+  if(!is.null(nmtran_presort_exe)){
+    # NM-TRAN input is now output from nmtran_presort
+    nmtran_input <- file.path(
+      paste0(fs::path_ext_remove(mod_path_new), "_presort.", fs::path_ext(mod_path_new))
+    )
+
+    processx::run(
+      command = nmtran_presort_exe,
+      stdin = mod_path_new,
+      stdout = nmtran_input,
+      wd = run_dir,
+      error_on_status = TRUE
+    )
+
+    # On Windows, nmtran_presort outputs the results to a fort.6 file instead of
+    # the designated stdout location.
+    # - Overwrite stdout with contents of fort.6 if on windows
```

Comment:
Perhaps tweak this to

```
# In our Window testing, nmtran_presort and NM-TRAN output write stdout to
# fort.6 instead of the locations specified in the processx::run() calls.
#
# See https://github.com/metrumresearchgroup/bbr/pull/705 for more discussion.
```